### PR TITLE
fix #234: Added method to fixup non json-spec compliant floats to make the resp…

### LINKED ISSUE #234: Produce Json-compliant results in serve (Nan)
When the forecast contain NaN values, Python's `json` library encodes this as `Nan` and not as a the string `"Nan"`. This is not compliant to the json spec and some parsers don't know how to handle these values.

### PR DESCRIPTION
**Pull request by dotgc**
Added method to fixup non json-spec compliant floats to make the response json compliant

*Issue #, if available:*
https://github.com/awslabs/gluon-ts/issues/234

*Description of changes:*
Calls a method to sanitize and fix non json-spec compliant floats before returning the response from `/invocations` endpoint

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/model/forecast.py [hunk 1]
```diff
@@ -318,7 +343,7 @@ def as_json_dict(self, config: "Config") -> dict:
         result = super().as_json_dict(config)
 
         if OutputType.samples in config.output_types:
-            result["samples"] = self.samples.tolist()
+            result["samples"] = Forecast.jsonify_floats(self.samples.tolist())
```
Comment (jaheba) [inline src/gluonts/model/forecast.py]: I'm not sure what the right place for this call is. Maybe we want to add it to the `shell` module instead to ensure that it is always called on the result?

File: src/gluonts/model/forecast.py [hunk 2]
```diff
@@ -188,7 +188,32 @@ def as_json_dict(self, config: "Config") -> dict:
         if OutputType.samples in config.output_types:
             result["samples"] = []
 
-        return result
+        return Forecast.jsonify_floats(result)
+
+    @staticmethod
+    def jsonify_floats(json_object):
```
Comment (jaheba) [inline src/gluonts/model/forecast.py]: I think this is not the right place for the method. Maybe, we could add it to some `util` module. And maybe it's also better to move it to `shell`.

File: test/model/test_forecast.py [hunk 3]
```diff
@@ -56,3 +57,26 @@ def percentile(value):
     assert forecast.prediction_length == 1
     assert len(forecast.index) == pred_length
     assert forecast.index[0] == pd.Timestamp(START_DATE)
+
+
+JSON_OBJECT_WITH_NON_COMPLIANT_FLOATS = {
```
Comment (jaheba) [inline test/model/test_forecast.py]: Maybe move this inside the test-method.

**Comment by szha** [comment]
Job PR-236/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-236/1/index.html

File: src/gluonts/model/forecast.py (hunk 1 above)
Comment (dotgc) [reply src/gluonts/model/forecast.py]: My initial thought was that `as_json_dict` method - given the method name, should be outputting well formed JSON and hence added it there, but it makes more sense to do it right before sending the response.

File: src/gluonts/model/forecast.py (hunk 2 above)
Comment (dotgc) [reply src/gluonts/model/forecast.py]: Done.

File: test/model/test_forecast.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 import pytest
 
 # First-party imports
-from gluonts.model.forecast import QuantileForecast, SampleForecast
+from gluonts.model.forecast import Forecast, QuantileForecast, SampleForecast
```
Comment (jaheba) [inline test/model/test_forecast.py]:
```suggestion
from gluonts.model.forecast import QuantileForecast, SampleForecast
```

This is not needed anymore.

File: test/model/test_forecast.py (hunk 3 above)
Comment (dotgc) [reply test/model/test_forecast.py]: Done.

File: test/model/test_forecast.py (hunk 4 above)
Comment (dotgc) [reply test/model/test_forecast.py]: Yep. Removed.

**Comment by szha** [comment]
Job PR-236/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-236/2/index.html

**Pull request merged**